clap_lex/src/ext.rs

```diff
@@ -287,6 +287,12 @@ mod private {
 /// There is no guarantee how non-UTF8 bytes will be encoded, even within versions of this crate
 /// (since its dependent on rustc)
 fn to_bytes(s: &OsStr) -> &[u8] {
+    // Unix and WASI have a safe API for these conversions.
+    #[cfg(unix)]
+    return std::os::unix::ffi::OsStrExt::as_bytes(s);
+    #[cfg(target_os = "wasi")]
+    return std::os::wasi::ffi::OsStrExt::as_bytes(s);
```

Comment:
I had considered this but decided against it so there was more consistency in what we are doing.  Granted, the underlying platform code is very inconsistent...

Reply:
It does basically the same thing (`from_bytes` even calls `transmute`) but it gives me more peace of mind. Seems best to err on the side of caution if it doesn't make the other code worse.

Reply:
My concern isn't how the code works its that I'd rather not shunt Windows support off into being the bad citizen but I'd rather be consistently bad.

Reply:
I guess it depends on your motivation for being consistently bad—I take it as a way to ensure that Windows doesn't end up neglected with a worse experience for end users. There's a risk that you start writing code that only works on Unix, or only works well there, with Windows half-functional or ignored or maybe impossible to support because of the feature's design. But maybe you mean something else?

I could add a comment saying that this is a safeguard, and not an excuse to write any more platform-dependent code.

Then if it makes a difference at all it'll be by preventing a lot of broken builds, which doesn't make things worse for Windows users.

Reply:
> I guess it depends on your motivation for being consistently bad—I take it as a way to ensure that Windows doesn't end up neglected with a worse experience for end users. There's a risk that you start writing code that only works on Unix, or only works well there, with Windows half-functional or ignored or maybe impossible to support because of the feature's design. But maybe you mean something else?

Its not for a slippery slope but for the here and now.

> Then if it makes a difference at all it'll be by preventing a lot of broken builds, which doesn't make things worse for Windows users.

What do you mean?

Reply:
Can it change?  Is there a way to write an `OsStr` implementation that is not effectively `transparent` for `[u8]`?

I'm having a hard time thinking of how it could be implemented any other way.  It is a [dynamically sized type](https://doc.rust-lang.org/reference/dynamically-sized-types.html) which puts its own set of restrictions on it but even more limiting is that any arbitrary `&str` can be converted into an `&OsStr` means that `OsStr` cannot own anything or really be anything else but a transmute of the `&str`.  If there is something I'm missing in this, I'm open to hearing it.

Reply:
Ah, yeah, I got that wrong—the DST restrictions do make it hard to break, especially at build time. The worst concrete scenario I can think of is that analyzers complain about what is technically already UB. But neither am I 100% sure that there is no other way for it to break.

I see it as a slim chance of a big headache that can be avoided at no cost. But I don't know anything you don't, so feel free to close at this point.